include/llbuild/Core/BuildEngine.h

```diff
@@ -431,4 +462,16 @@ class BuildEngine {
 }
 }
 
+namespace std
+{
+  template<>
+  struct hash<llbuild::core::KeyType>
+  {
+    size_t operator()(const llbuild::core::KeyType& key) const
+    {
+      return hash<std::string>{}(key.function);
```

Comment:
This doesn't seem correct, to use function & args in == and < but not in the hash.